packages/server/lib/routes.ts

```diff
@@ -65,12 +69,15 @@ export const createCommonRoutes = ({
   router.get(clientRoute, (req, res) => {
     debug('Serving Cypress front-end by requested URL:', req.url)
 
+    // ensure the remote state gets cleaned up from any previous tests/runs
+    remoteStates.reset()
+
```

Comment:
it's weird to make `GET` have a side-effect like wiping out `remoteStates`, is there a better place we could do this? like maybe in onResetServerState only? I see we're already doing it there: https://github.com/cypress-io/cypress/blob/890bde6a053525dfaaac9abcdf041c98753d50c6/packages/server/lib/server-base.ts#L331-L335

Reply:
This reset handles resetting in-between tests to ensure we start clean on each test. However, if the browser is closed/crashes/refreshed in the middle of the `cy.origin` command, the server and front-end can become out of sync. That's where the other resets come into play. I can see if there is another location to do those resets or if you know of one, let me know.

Reply:
Updated [7c8d698](https://github.com/cypress-io/cypress/pull/21137/commits/7c8d698dd8885c4a872c507ebc571e94a234310f)